projects/arlas-toolkit/src/lib/components/histogram-tooltip-overlay/histogram-tooltip-overlay.component.html

```diff
@@ -1,40 +1,51 @@
+     <span class="interval-text" *ngIf="interval.end">to</span>
+     <ng-container *ngTemplateOutlet="intervalTemplate; context: {value: interval.end, isTime: tooltip.dataType === 'time' }"></ng-container>
+     <ng-container *ngTemplateOutlet="intervalTextTemplate;
+      context: {unit:tooltip.xUnit, interval:tooltip.xRange.unit, value: tooltip.xRange.value, isTime: tooltip.dataType === 'time'  }"></ng-container>
   </div>
-  <div class="xline">
-    <div class="labelwrapper">
-      <div class="axis">{{'x-axis:' | translate}}</div>
-    </div>
-    <div style="display: flex;">
-      <div class="xlabel">{{tooltip.xLabel | translate}}</div>
-      <div>{{(tooltip?.dataType === 'time' ? 'From' : '') | translate}} {{tooltip.xValue}} {{tooltip.xUnit}}</div>
-    </div>
+   <div class="tooltip__yLabel label">
+     {{tooltip.yLabel | translate}} <span class="axis-label">({{'y-axis' | translate}})</span>
   </div>
-
-  <div class="yline">
-    <div class="labelwrapper">
-      <div class="axis">{{'y-axis:' | translate}}</div>
-    </div>
-    <table class="values">
-      <tr *ngFor="let y of tooltip.y">
-        <td>
-          <div class="chip" [style.backgroundColor]="y.color"></div>
-        </td>
-        <td class="count">
-          <div>{{y.value}}</div>
-        </td>
-        <td  *ngIf="tooltip?.dataType === 'time'">
-          <div class="value_unit">{{y.chartId}}</div>
-        </td>
-        <td  *ngIf="tooltip?.dataType !== 'time' && tooltip.yUnit !== undefined && tooltip.yUnit !== ''">
-          <div class="value_unit">{{tooltip.yUnit}}</div>
-        </td>
-      </tr>
-    </table>
+   <div class="tooltip__yValue value-label">
+     <table class="values">
+       <tr *ngFor="let y of tooltip.y">
+         <td>
+           <div class="chip" [style.backgroundColor]="y.color"></div>
+         </td>
+         <td class="count">
+           <div>{{y.value}}</div>
+         </td>
+         <td  *ngIf="tooltip?.dataType === 'time'">
+           <div class="value-unit">{{(tooltip.yUnit && tooltip.yUnit .length > 0)?  tooltip.yUnit : y.chartId}}</div>
+         </td>
+         <td  *ngIf="tooltip?.dataType !== 'time' && tooltip.yUnit !== undefined && tooltip.yUnit !== ''">
+           <div class="value-unit">{{tooltip.yUnit}}</div>
+         </td>
+       </tr>
+     </table>
   </div>
 </div>
+
+<!-- template for x unit -->
+<ng-template #intervalTemplate let-value="value" let-isTime="isTime">
+  <ng-container *ngIf="isTime else notTemporal">
+    {{value| date:"dd/MM/YYYY"}}
+  </ng-container>
+   <ng-template #notTemporal>
+     {{value}}
+   </ng-template>
+</ng-template>
+
+
+<ng-template #intervalTextTemplate let-interval="interval" let-unit="unit" let-value="value" let-isTypeTime="isTime">
+  <span class="interval-text" *ngIf="isTypeTime">{{value}}</span>
+  <span class="interval-text" *ngIf="!isTypeTime && unit">{{unit | translate}}</span>
+  <span class="interval-text" *ngIf="interval">{{interval| translate}}</span>
```

Comment:
when it's not temporal, we should write down:

({{'interval' | translate} } : {{interval| translate}})